controllers/machine_controller.go

```diff
@@ -183,9 +185,15 @@ func (r *MachineReconciler) reconcileDelete(ctx context.Context, cluster *cluste
 		}
 	} else {
 		klog.Infof("Deleting node %q for machine %q", m.Status.NodeRef.Name, m.Name)
-		if err := r.deleteNode(ctx, cluster, m.Status.NodeRef.Name); err != nil && !apierrors.IsNotFound(err) {
-			klog.Errorf("Error deleting node %q for machine %q: %v", m.Status.NodeRef.Name, m.Name, err)
-			return ctrl.Result{}, err
+		waitErr := wait.PollImmediate(2*time.Second, 10*time.Second, func() (bool, error) {
+			if err = r.deleteNode(ctx, cluster, m.Status.NodeRef.Name); err != nil && !apierrors.IsNotFound(err) {
+				return false, nil
+			}
+			return true, nil
+		})
+		if waitErr != nil {
+			r.Log.Error(err, "timed out deleting Machine's node, moving on", "node", m.Status.NodeRef.Name, "machine", m.Name)
```

Comment:
Is the machine name needed here in the log message? Shouldn't the logger already be configured with the machine name using WithValues at this point?

Reply:
Should be, but there's a bug in the code :-(

https://github.com/kubernetes-sigs/cluster-api/blob/90cbdd174b1753926e26ef1393356d030139e176/controllers/machine_controller.go#L78

Reply:
:grimacing:

Reply:
I'm now working on https://github.com/kubernetes-sigs/cluster-api/issues/1203
lets fix this on that pr?

Reply:
SGTM. Please add a TODO in this PR to remove the machine from this line